pkg/infrastructure/azure/azure.go

```diff
@@ -937,3 +943,16 @@ func getMachinePoolSecurityType(in clusterapi.InfraReadyInput) (string, error) {
 	}
 	return "", nil
 }
+
+func getMachinePoolZones(in clusterapi.PreProvisionInput) ([]string, error) {
+	var mpool *aztypes.MachinePool
+	if in.InstallConfig.Config.ControlPlane != nil && in.InstallConfig.Config.ControlPlane.Platform.Azure != nil {
+		mpool = in.InstallConfig.Config.ControlPlane.Platform.Azure
+	} else if in.InstallConfig.Config.Platform.Azure.DefaultMachinePlatform != nil {
+		mpool = in.InstallConfig.Config.Platform.Azure.DefaultMachinePlatform
+	}
+	if mpool == nil {
+		return []string{}, fmt.Errorf("failed to get machine pool")
+	}
```

Comment:
this seems to be just looking at the install config, i.e. this will only show multiple zones if users specify zones in the install config, but generally users just specify a region and the installer picks the zones for them. You could potentially solve this issue like this:

https://github.com/openshift/installer/blob/99b9ef1100a4685c73b55f3523d98718b72512f2/pkg/asset/machines/master.go#L348-L349

Or check the availability zone of each machine manifest and see if there are multiple zones, which would be using the result of this earlier call.

Reply:
This as written works and is tested. I will see how this works if configuring zones manually in the install-config. If someone does that and specifies more than 1 zone, this will still work.

Reply:
Oh, I see. 

> This gets saved here:
> 
> https://github.com/openshift/installer/blob/a4449b336a62cd43cc565c1db6692cd4dedb1e1b/pkg/asset/machines/clusterapi.go#L273


I missed that this PR saves the machine pool back into the `InstallConfig` asset. I suspect this will work as long as the `InstallConfig` asset is held in memory, but would break if using a multistage install, e.g. `create manifests` then `create cluster`

Reply:
Tested using manifest creation and then install and it works as expected.

Reply:
The assets get written to disk which writes out the zones